Add metrics collection and metrics-based dashboards info to HAProxy Otel app (#4265)
SumoLogic/sumologic-documentation: docs/integrations/web-servers/opentelemetry/haproxy-opentelemetry.md

<img src={useBaseUrl('img/integrations/web-servers/haproxy.png')} alt="Thumbnail icon" width="50"/> <img src={useBaseUrl('img/send-data/otel-color.svg')} alt="Thumbnail icon" width="45"/>

> [HAProxy](https://docs.haproxy.org/2.6/intro.html) app is a unified logs and metrics app that helps you monitor the availability, performance, health, and resource utilization of HAProxy server farms. Preconfigured dashboards and searches provide visibility into your environment for real-time or historical analysis: visitor locations, HTTP error codes percentage, backend and frontend server statistics, traffic patterns, errors, server operations, and access from known malicious sources.

Reviewer: ```suggestion
[HAProxy](https://docs.haproxy.org/2.6/intro.html) app is a unified logs and metrics app designed to help monitor the availability, performance, health, and resource utilization of HAProxy server farms. It provides preconfigured dashboards and searches that offer visibility into various aspects of your environment for both real-time and historical analysis: visitor locations, HTTP error codes percentage, backend and frontend server statistics, traffic patterns, errors, server operations, and access from known malicious sources.
```

> The [HAProxy](https://docs.haproxy.org/2.6/intro.html) app is a unified logs and metrics app designed to help monitor the availability, performance, health, and resource utilization of HAProxy server farms. It provides preconfigured dashboards and searches that offer visibility into your environment for both real-time and historical analysis: visitor locations, HTTP error codes percentage, backend and frontend server statistics, traffic patterns, errors, server operations, and access from known malicious sources.

Reviewer 2: ```suggestion
The [HAProxy](https://docs.haproxy.org/2.6/intro.html) app is a unified logs and metrics app designed to help you monitor the availability, performance, health, and resource utilization of HAProxy server farms. It provides preconfigured dashboards and searches that offer visibility into your environment for real-time and historical analysis: visitor locations, HTTP error codes percentage, backend and frontend server statistics, traffic patterns, errors, server operations, and access from known malicious sources.
```

Author: fixed it.

> The OpenTelemetry collector runs on the same host as HAProxy and uses the [HAProxy Receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/haproxyreceiver) and the [Sumo Logic OpenTelemetry Exporter](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/sumologicexporter) to send the metrics to Sumo Logic.HAProxy logs are sent to Sumo Logic through OpenTelemetry [filelog receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/filelogreceiver).

Reviewer: ```suggestion
The OpenTelemetry collector runs on the same host as HAProxy and uses the [HAProxy Receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/haproxyreceiver) and the [Sumo Logic OpenTelemetry Exporter](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/sumologicexporter) to send the metrics to Sumo Logic. HAProxy logs are sent to Sumo Logic through OpenTelemetry [filelog receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/filelogreceiver).
```

Author: fixed it.

> The OpenTelemetry collector runs on the same host as HAProxy and uses the [HAProxy Receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/haproxyreceiver) and the [Sumo Logic OpenTelemetry Exporter](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/sumologicexporter) to send the metrics to Sumo Logic. HAProxy logs are sent to Sumo Logic through the OpenTelemetry [filelog receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/filelogreceiver).

Reviewer 2: ```suggestion
The OpenTelemetry collector runs on the same host as HAProxy, where it uses the [HAProxy Receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/haproxyreceiver) and the [Sumo Logic OpenTelemetry Exporter](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/sumologicexporter) to send the metrics to Sumo Logic. HAProxy logs are sent to Sumo Logic through the OpenTelemetry [filelog receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/filelogreceiver).
```

Author: fixed it.

Following are the [Fields](/docs/manage/fields/) which will be created as part of HAProxy App install if not already present.

> - **`webengine.cluster.name`**. User configured. Enter a name to identify the Haproxy cluster. This cluster name will be shown in the Sumo Logic dashboards.

Reviewer 3: Can you add deployment.environment here

> - **`webengine.cluster.name`**. User configured. Enter a name to identify the Haproxy cluster. This cluster name will be shown in the Sumo Logic dashboards.

Reviewer: ```suggestion
- **`webengine.cluster.name`**. User configured. Enter a name to identify the HAProxy cluster. This cluster name will be shown in the Sumo Logic dashboards.
```

Author: fixed it.

- **`webengine.system`**. Has fixed value of **haproxy**
- **`sumo.datasource`**. Has fixed value of **haproxy**
- **`webengine.node.name`**. Has the value of host name of the machine which is being monitored

## Prerequisites

### For metrics collection

> The receiver used gets stats from an HAProxy instance using the `stats` endpoint. This receiver supports HAProxy version 2.3.9+.

Reviewer: ```suggestion
The receiver used gets stats from an HAProxy instance using the `stats` endpoint. This receiver supports HAProxy version 2.3.9 and later.
```

Author: fixed it.

Following are the [Fields](/docs/manage/fields/) which will be created as part of HAProxy App install if not already present.

> - **`sumo.datasource`**. Has fixed value of **haproxy**

Reviewer 2: ```suggestion
- **`sumo.datasource`**. Has fixed value of **haproxy**.
```

Author: fixed it.

> - **`webengine.system`**. Has fixed value of **haproxy**

Reviewer 2: ```suggestion
- **`webengine.system`**. Has fixed value of **haproxy**.
```

Author: fixed it.

- **`webengine.cluster.name`**. User configured. Enter a name to identify the HAProxy cluster. This cluster name will be shown in the Sumo Logic dashboards.

> - **`webengine.node.name`**. Has the value of host name of the machine which is being monitored

Author: fixed it.

> - **`deployment.environment`**. User configured. This is the deployment environment where the Memcache cluster resides. For example: dev, prod or qa.

Author: fixed it.

In this step, you will configure the yaml required for HAProxy Collection.

Below are the inputs required:

- **Endpoint**. The URL of the httpd status endpoint (default: `http://localhost:8404/stats`).
- **HAProxy logs Path**. Enter the path to the log file for your HAProxy instance.

> - **Fields**. `webengine.cluster.name`.

Reviewer 3: This line is not making any sense. Can we remove it or provide more information

Author: removed it.

> The path of the log file configured to capture haproxy logs is needed to be given here.

Reviewer 3: This is already covered above under HAProxy log Path

Author: fixed it.

In this step, you will configure the yaml required for HAProxy Collection.

Below are the inputs required:

- **Endpoint**. The URL of the httpd status endpoint (default: `http://localhost:8404/stats`).
- **HAProxy logs Path**. Enter the path to the log file for your HAProxy instance.

> The path of the log file configured to capture haproxy logs is needed to be given here. The files are typically located in `/var/log/haproxy*.log`. If you're using a customized path, check the haproxy.conf file for this information. You can add any custom fields which you want to tag along with the data ingested in Sumo. Click on the **Download YAML File** button to get the yaml file.

Reviewer 2: ```suggestion
The path of the log file configured to capture haproxy logs is needed to be given here. The files are typically located in `/var/log/haproxy*.log`. If you're using a customized path, check the `haproxy.conf` file for this information. You can add any custom fields which you want to tag along with the data ingested in Sumo Logic. Click on the **Download YAML File** button to get the yaml file.
```

Author: fixed it.

| limit 5
```

### Metrics

> Here's a sample Metrics query from the **Http Response Codes** dashboard > **HAProxy - Backend Metrics** panel:

Reviewer 2: ```suggestion
Here is a sample metrics query from the **Http Response Codes** dashboard > **HAProxy - Backend Metrics** panel:
```

Author: file updated.

| limit 5
```

### Metrics

Here's a sample Metrics query from the **Http Response Codes** dashboard > **HAProxy - Backend Metrics** panel:

```

> sumo.datasource=haproxy metric=haproxy.requests.total status_code=* haproxy.service_name=backend deployment.environment={{deployment.environment}} webengine.cluster.name={{webengine.cluster.name}} webengine.node.name={{webengine.node.name}}  haproxy.proxy_name={{haproxy.proxy_name}} 

Reviewer 3: this should be a resolved query. That is all the parameters which are inside {{}} should be replaced with *

Author: fixed it.

| parse field=status_code *  as code 
| avg by webengine.cluster.name,webengine.node.name,haproxy.proxy_name,code
```

> ## Sample Metrics

Reviewer: ```suggestion
## Sample metrics
```

Author: fixed it.

<img src='https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/HAProxy-OpenTelemetry/HAProxy-Overview.png' alt="Overview" />

### Backend

> The **HAProxy - Backend** dashboard provides an at-a-glance view for the number of backend active servers, backend weight, respond code from backend and throughput http.

Author: changes updated.

<img src='https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/HAProxy-OpenTelemetry/HAProxy-Backend.png' alt="Error Log Analysis" />

### Frontend

> The **HAProxy - Backend** dashboard provides an at-a-glance view detail of HAProxy Frontend. It provides information such as number request to frontend, number of error requests, and current session.

Reviewer 2: ```suggestion
The **HAProxy - Frontend** dashboard provides details of HAProxy Frontend. It provides information such as number request to frontend, number of error requests, and current session.
```

Author: updated changes.

<img src='https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/HAProxy-OpenTelemetry/HAProxy-Frontend.png' alt="Error Log Analysis" />

### Server

> The **HAProxy - Backend** dashboard provides an at-a-glance view detail of HAProxy Server. This dashboard helps you monitoring uptime, and error request by proxy.

Reviewer 2: ```suggestion
The **HAProxy - Server** dashboard provides details of HAProxy Server. This dashboard helps you monitor the uptime and error request by proxy.
```

Author: changes updated.
